### ISSUE TITLE
dotnet ef dbcontext scaffold on Mac creates line ending differences

### ISSUE THREAD

**Issue by tpitman** [dotnet/efcore#29948]
### The Command

When I run the following command on Windows, any lines that have a blank line end with CRLF CRLF

```
dotnet ef dbcontext scaffold ...
```

When I run that same command on a Mac, any lines that have a blank line end with CRLF LF

I saw this because git kept showing changes in the files on those types of lines. Any lines that have something on them, seem to be fine. It is only those lines that are blank have just the LF as the blank.

Those lines should behave like on Window where all lines end in CRLF.

EDIT: As roji pointed out below I shouldn't have said they should behave like Windows. They should behave consistently, however, and if the tool is in control of this, then maybe make it consistent across platforms and always make all lines end in CRLF? If not, then always do LF only on mac, but NOT mixed. Then I can control it via git.

### Versions

EF Core version: 7.0.1
Database provider: Npgsql.EntityFrameworkCore.PostgreSQL
Target framework: .NET 7.0
Operating system: MacOS 12.6
IDE: Visual Studio 2022 17.4

### Images

<img width="407" alt="image" src="https://user-images.githubusercontent.com/2906563/210082392-25542ecd-0452-4648-a875-d1838bf3b23b.png">

<img width="955" alt="image" src="https://user-images.githubusercontent.com/2906563/210082638-074f6ff8-f1b9-464f-ac5f-d4aeeaca7dc6.png">

**Comment by roji** [comment]
> Those lines should behave like on Window where all lines end in CRLF.

Different platforms have different line endings, why should EF generate files on Mac with Windows line endings? By design, EF generates line endings based on the platform where iti s executed.

If developers are working on different platforms, then the usual way to normalize line endings is when committing e.g. to git ([see e.g. this](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings)).

**Comment by tpitman** [comment]
I understand what you are saying, but it doesn't line up with what is happening.

If ALL lines endings where either CRLF or LF only then I wouldn't be brining this up, but that isn't what is happening.

If you look at the second screenshot you can see the mixture and only when there is a blank line.

For some reason the ef tool is generating a MIXTURE of line endings, so they obviously have some control over it. If they would pick on or the other then that would be great, but mixing them is not good.

I suspect they are putting CRLF at the end of all lines that have text on them, but when encountering a line without any text they have a special ELSE or CASE that puts the LF only.

**Comment by ajcvickers** [comment]
Note from triage: could be due to use of the templates for generation.

**Comment by bricelam** [comment]
> Note from triage: could be due to use of the templates for generation.

Correct. The T4 pre-processor injects CRLF into the string literals. We could normalize the line endings after transforming, but this might cause line endings in SQL (e.g. DEFAULT constraints, index filters, etc.) to be changed.

**Comment by bricelam** [comment]
I bet you can work around this be adding the default templates to your project and changing their line endings.

**Comment by shuebner** [comment]
Also, don't forget there may be an `.editorconfig` in scope that demands `LF` line endings, independent of OS.

It is a bit clumsy to scaffold with a tool that ignores those settings and then to have to change the line-endings afterwards.

**Comment by miroljub1995** [comment]
We get CRLF generated on linux too.
